frontend/src/utils/api.ts

```diff
@@ -16,7 +16,11 @@ api.interceptors.response.use(
       const currentPath = window.location.pathname;
       const sourcePage = encodeURIComponent(currentPath);
       localStorage.removeItem("token");
-      window.location.href = `/login?sourcePage=${sourcePage}`;
+      if (currentPath.includes("/login")) {
```

Comment:
Was just redirecting instead of throwing and error when logging in with bad credentials